tests/eas/generic.py

```diff
@@ -252,9 +256,80 @@ def exp_power(row):
             power = nrg_model.estimate_from_cpu_util(expected_utils[0])
             columns = power.keys()
             return pd.Series([power[c] for c in columns], index=columns)
+
+        self.plot_expected_util(experiment)
+
         return self._sort_power_df_columns(
             task_utils_df.apply(exp_power, axis=1))
 
+    def plot_expected_util(self, experiment):
+        """
+        Create a plot of the expected per-CPU utilization for the experiment
+        The plot is then outputted to the test results directory.
+
+        :param experiment: The :class:Experiment to examine
+        :returns: A Pandas DataFrame with a column each node in the energy model
+                  (keyed with a tuple of the CPUs contained by that node) and a
+                  "power" column with the sum of other columns. Shows the
+                  estimated *optimal* power over time.
+        """
+        task_utils_df = self.get_task_utils_df(experiment)
+        nrg_model = self.te.nrg_model
+        trace = self.get_trace(experiment)
+
+        cpus = nrg_model.cpus
+        data = []
+        index = []
+        for time, utils_row in task_utils_df.iterrows():
+            task_utils = utils_row.to_dict()
+            expected_utils = nrg_model.get_optimal_placements(task_utils)[0]
+
+            row = []
+            for cpu, util in enumerate(expected_utils):
+                row.append(util)
```

Comment:
Isn't this just `row = expected_utils`?

Reply:
Totally is yeah, must've tried something fancy with cpu indexing and eventually gave up.